### ISSUE TITLE
Set TTL for connection pool

### ISSUE THREAD

**Issue by sagebind** [sagebind/isahc#93]
Building on top of #81, we should allow users to customize and set a max timeout that inactive connections should be kept in the connection pool ([`CURLOPT_MAXAGE_CONN`](https://curl.haxx.se/libcurl/c/CURLOPT_MAXAGE_CONN.html)).

Note that this option is very new and will not work if we don't use the static bundled curl version.

### Proposed API

```rust
impl HttpClientBuilder {
    /// Set the maximum time-to-live (TTL) for connections to remain in the
    /// connection cache.
    ///
    /// After requests are completed, if the underlying connection is
    /// reusable, it is added to the connection cache to be reused to reduce
    /// latency for future requests. This option controls how long such
    /// connections should be still considered valid before being discarded.
    ///
    /// Old connections have a high risk of not working any more and thus
    /// attempting to use them wastes time if the server has disconnected.
    ///
    /// The default TTL is 118 seconds.
    pub fn connection_cache_ttl(self, ttl: Duration) -> Self;
}
```

### Implementation notes

If the `CURLOPT_MAXAGE_CONN` option isn't exposed by the [upstream curl crate](https://github.com/alexcrichton/curl-rust), we can first add it to that crate (preferred) or use FFI directly to invoke it ourselves.

While `CURLOPT_MAXAGE_CONN` _can_ be configured differently for every easy handle, it can produce really strange results when running multiple requests concurrently using a multi handle since the connection pool is shared between all requests. This is similar to [`dns_cache`](https://github.com/sagebind/isahc/blob/e9c54c5abcc64972019786ac2bfdbc7b82f133d6/src/client.rs#L220-L227), which while technically curl supports setting different values per request, we do not expose it because it results in surprising behavior.

**Comment by gsquire** [comment]
I submitted a PR to the curl bindings to add this constant in. After that is merged, I'd be happy to take this issue.

**Comment by gsquire** [comment]
Now that the upstream PR is merged, do you want to wait until a new patch version is released to make this change?

**Comment by sagebind** [comment]
Yeah, I'll push for a new version to get published today or tomorrow. In the meantime you can also reference the latest Git `master` branch in `Cargo.toml` in order to work on the feature until the patch is released.

**Comment by sagebind** [comment]
Version [0.4.34](https://github.com/alexcrichton/curl-rust/releases/tag/0.4.34) of the curl crate has been published with your updates, so we can now use it from here.

**Comment by sagebind** [comment]
This is now available in the [0.9.11 release](https://github.com/sagebind/isahc/releases/tag/0.9.11)!
